src/coreComponents/physicsSolvers/multiphysics/CoupledSolver.hpp

```diff
@@ -307,20 +332,194 @@ class CoupledSolver : public SolverBase
+  {
+    GEOSX_MARK_FUNCTION;
+
+    real64 dtReturn = dt;
+
+    // setup the coupled linear system
+    setupSystem( domain, m_dofManager, m_localMatrix, m_rhs, m_solution );
+
+    // setup reservoir and well systems
+    implicitStepSetup( time_n, dt, domain );
+
+    // currently the only method is implicit time integration
+    dtReturn = nonlinearImplicitStep( time_n, dt, cycleNumber, domain );
+
+    // complete time step
+    implicitStepComplete( time_n, dtReturn, domain );
+
+    return dtReturn;
+  }
+  /**
+   * @brief Sequentially coupled solver step. It solves a nonlinear system of
+   * equations using a sequential approach.
+   *
+   * @param time_n the current time
+   * @param dt timestep size
+   * @param cycleNumber
+   * @param domain the domain partition
+   * @return real64 size of the accepted timestep
+   */
+  virtual real64 sequentiallyCoupledSolverStep( real64 const & time_n,
+                                                real64 const & dt,
+                                                int const cycleNumber,
+                                                DomainPartition & domain )
+  {
+    GEOSX_MARK_FUNCTION;
+
+    real64 dtReturn = dt;
+
+    real64 dtReturnTemporary;
+
+    forEachArgInTuple( m_solvers, [&]( auto & solver, auto )
+    {
+      solver->setupSystem( domain,
+                           solver->getDofManager(),
+                           solver->getLocalMatrix(),
+                           solver->getSystemRhs(),
+                           solver->getSystemSolution() );
+
+      solver->implicitStepSetup( time_n, dt, domain );
+
+    } );
+
+    // need to check what to do with this. this->implicitStepSetup( time_n, dt, domain );
+    NonlinearSolverParameters & solverParams = getNonlinearSolverParameters();
+    integer & iter = solverParams.m_numNewtonIterations;
+    iter = 0;
+    bool isConverged = false;
+    /// Sequential coupling loop
+    while( iter < solverParams.m_maxIterNewton )
+    {
+      if( iter == 0 )
+      {
+        // Reset the states of all solvers if any of them had to restart
+        forEachArgInTuple( m_solvers, [&]( auto & solver, auto )
+        {
+          solver->resetStateToBeginningOfStep( domain );
+        } );
+        resetStateToBeginningOfStep( domain );
+      }
+
+      forEachArgInTuple( m_solvers, [&]( auto & solver, auto idx )
+      {
+        GEOSX_LOG_LEVEL_RANK_0( 1, GEOSX_FMT( "  Iteration {:2}: {}", iter+1, solver->getName() ) );
+        dtReturnTemporary = solver->nonlinearImplicitStep( time_n,
+                                                           dtReturn,
+                                                           cycleNumber,
+                                                           domain );
+
+        mapSolutionBetweenSolvers( domain, idx() );
+
+        if( dtReturnTemporary < dtReturn )
+        {
+          iter = 0;
+          dtReturn = dtReturnTemporary;
+        }
```

Comment:
Later, we can add an option that will force the coupled solver to cut the time step if one of the solvers has a time step failure. Just to make sure that we have full control on the behavior of the outer iteration.

Reply:
yeah, it's actually the only reason why I had set the default `subcycling` to 0 for now.